k2/csrc/top_sort.cu

```diff
@@ -0,0 +1,533 @@
+/**
+ * @brief
+ * compose
+ *
+ * @copyright
+ * Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#include <limits>
+#include <vector>
+
+#include "k2/csrc/array_ops.h"
+#include "k2/csrc/fsa_algo.h"
+
+namespace k2 {
+
+// Caution: this is really a .cu file.  It contains mixed host and device code.
+
+/*
+   Pruned intersection (a.k.a. composition) that corresponds to decoding for
+   speech recognition-type tasks.  Can use either different decoding graphs (one
+   per acoustic sequence) or a shared graph
+*/
+class TopSorter {
+ public:
+  /**
+     Topological sorter object.  You should call TopSort() after
+     constructing it.
+
+       @param [in] fsas    A vector of FSAs; must have 3 axes.
+   */
+  TopSorter(FsaVec &fsas)
+    : c_(fsas_.Context()), fsas_(fsas) {
```

Comment:
You probably meant fsas.Context() instead of fsas_.Context() since fsas_ is not initialized yet at this point?

Reply:
yes thx